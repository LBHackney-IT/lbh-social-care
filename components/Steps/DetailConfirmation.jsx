import PropTypes from 'prop-types';
import { useRouter } from 'next/router';
import Button from 'components/Button/Button';

const DetailConfirmation = ({ formSteps, successMessage }) => {
  const router = useRouter();
  const { id } = router.query;
  if (!formSteps) return null;
  return (
    <>
      <h1 className="govuk-fieldset__legend--l gov-weight-lighter govuk-!-margin-bottom-7">
        {successMessage || 'Submission complete'}
      </h1>
      <div className="govuk-panel govuk-panel--confirmation govuk-!-margin-bottom-9">
        <h1 className="govuk-fieldset__legend--l gov-weight-lighter">
          {formSteps[0].title} details have been added
        </h1>
      </div>
      <div style={{ display: 'flex', justifyContent: 'space-between' }}>
        <Button label="Return to search" isSecondary wideButton route={`/`} />
        <Button label="View person" wideButton route={`/people/${id}`} />
      </div>
    </>
  );
};

DetailConfirmation.propTypes = {
  formData: PropTypes.shape({}).isRequired,
  formSteps: PropTypes.arrayOf(PropTypes.shape({ title: PropTypes.string }))
    .isRequired,
  formPath: PropTypes.string.isRequired,
  successMessage: PropTypes.string,
  isSummaryCollapsable: PropTypes.bool,
};

export default DetailConfirmation;
