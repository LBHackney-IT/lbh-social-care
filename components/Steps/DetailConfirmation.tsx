import { useRouter } from 'next/router';

import Button from 'components/Button/Button';
import { FormStep } from 'components/Form/types';

interface Props {
  formSteps: FormStep[];
  successMessage?: string;
}

const DetailConfirmation = ({
  formSteps,
  successMessage,
}: Props): React.ReactElement => {
  const router = useRouter();
  const { id } = router.query;
  return (
    <>
      <h1 className="govuk-fieldset__legend--l gov-weight-lighter govuk-!-margin-bottom-7">
        {successMessage || 'Submission complete'}
      </h1>
      <div className="govuk-panel govuk-panel--confirmation govuk-!-margin-bottom-9">
        <h1 className="govuk-fieldset__legend--l gov-weight-lighter">
          {formSteps[0].title} details have been added
        </h1>
      </div>
      <div style={{ display: 'flex', justifyContent: 'space-between' }}>
        <Button label="Return to search" isSecondary wideButton route={`/`} />
        <Button label="View person" wideButton route={`/people/${id}`} />
      </div>
    </>
  );
};

export default DetailConfirmation;
