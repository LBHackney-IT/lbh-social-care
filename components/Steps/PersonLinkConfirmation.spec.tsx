import { render } from '@testing-library/react';

import DetailConfirmation from './PersonLinkConfirmation';

jest.mock('next/router', () => ({
  useRouter: () => ({
    query: { id: 'foo123456bar' },
  }),
}));

describe('Detail Confirmation component', () => {
  const props = {
    successMessage: 'Done!',
  };

  it('should render properly', () => {
    const { asFragment, getByText } = render(<DetailConfirmation {...props} />);
    expect(asFragment()).toMatchSnapshot();
    expect(getByText(props.successMessage)).toBeInTheDocument();
  });
});
