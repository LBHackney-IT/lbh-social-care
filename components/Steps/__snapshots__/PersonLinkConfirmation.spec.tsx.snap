// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Confirmation component should render properly 1`] = `
<DocumentFragment>
  <div
    class="govuk-panel govuk-panel--confirmation lbh-panel govuk-!-margin-bottom-9"
  >
    <h1
      class="govuk-panel__title"
    >
      Done!
    </h1>
  </div>
  <div
    style="display: flex; justify-content: space-between;"
  >
    <button
      class="lbh-button govuk-button lbh-button--secondary"
      data-module="govuk-button"
      style="min-width: 200px;"
    >
      Return to search
    </button>
    <button
      class="lbh-button govuk-button"
      data-module="govuk-button"
      style="min-width: 200px;"
    >
      View person
    </button>
  </div>
</DocumentFragment>
`;
